### ISSUE TITLE
artoolkitX Unity package and long black screen.

### ISSUE THREAD

**Issue by 5y** [artoolkit/arunity5#194]
Hello,
Regarding my problems with long black screen ( for getting camera calibration params from Artoolkit servers), I've tried to create a Unity package with new servers. However, it doesn't work for me, Hence, I have two requests, please.

1- Release a standard Unity package for [artoolkitx](http://www.artoolkitx.org/docs/downloads/). 

2-  It would be great if you inform me any possible solution regarding my problems and stop checking camera calibration params from servers. It would be perfect if we control this module on Unity, not in the binary packages. the long black screen is so annoying. I've tried some methods for disabling this function and it worked. However, I need a standard procedure for disabling this function and use the default params.

Thank you so much.
https://github.com/artoolkit/arunity5/issues/119#issuecomment-361015465
https://github.com/artoolkit/artoolkit5/issues/347#issuecomment-361048354

**Comment by 5y** [comment]
@ThorstenBux

**Comment by 5y** [comment]
@ThorstenBux Thank you.
I'm trying to disable cparamSearch.c
It would be great if you inform me which part of cparamSearch.c should be disabled. I found some clues. However, there is a connection between cparamSerach functions.

https://github.com/artoolkit/artoolkit5/blob/master/lib/SRC/Video/cparamSearch.c

**Comment by 5y** [comment]
@ThorstenBux I've got a copy of ARtoolkit unity repo but I can't find cparamSearch.c or any related items for getting params from the server.
you sent me a manual for creating Unity package from Artoolkit Unity repo. However, there is no cparamSearch.c function on Unity Artoolkit repo. Could you please send me a manual for creating unity package from the Artoolkit main repo or let me know how to find cparamSearch.c in the Unity ARtoolkit repo?

Thank you.

**Comment by rafaazvdo** [comment]
Hello @5y @ThorstenBux , i am with same problem. But i want only block the internet acess in my aplication. Have some way to do this in the Unity? I already tried change the AndroidManifest.xml, but i not got succes.

**Comment by 5y** [comment]
@rafaazvdo  The simple answer is you can not. You should shut down entire Unity Internet connection ( There is no solution for this method) or use the new version of Artoolkit with the new server (there is no Unity package for this version). I've changed this part of the code (cparamSearch.C) but I can't find a manual for creating Unity package from the main ARtoolkit repo. I'm waiting for @ThorstenBux @philip-lamb response.

**Comment by ThorstenBux** [comment]
You can rebuild the Unity plugin using the documentation here: https://github.com/artoolkit/arunity5/tree/master/src/Android_Unity_Player_Source/AndroidStudioProj

**Comment by 5y** [comment]
[@ThorstenBux]
Thank you so much for getting back to me.
I'm not sure this is a right manual. I tried this way. However, it is a manual for creating an ARtoolkit **Unity package** from ARtoolkit **Unity repository**. 

As you know,  you can find  cparamSearch.c is on the **main repository** of ARtoolkit (https://github.com/artoolkit/artoolkit5/blob/master/lib/SRC/Video/cparamSearch.c). 

The point of creating a new **Unity package** is disabling cparamSearch.c or even change the wrong server address But I can't access this function via ARtoolkit **Unity package repository.** I should change **cparamSearch.c on the main repository then create a unity package**. Do you have any idea?


Thanks./

**Comment by ThorstenBux** [comment]
My bad @5y 

you need to change in the main repo and then create the android libs android/build.sh then take the outcome of armeabi-v7a and x86 and copy that to your Unity repo at `src/Unity/Assets/Plugins/Android/libs/` that should do the trick

**Comment by 5y** [comment]
[@ThorstenBux] 

Thank you. 
So, armeabi-v7a and x86 should be sufficient for creating a multiplatfrom export from unity?

**Comment by ThorstenBux** [comment]
Yes it is. These are the only two architectures Unity3D supports for Android.

**Comment by 5y** [comment]
@ThorstenBux Sure, Thank you.
What's about iOS? Should I repeat the same method for creating iOS libs?

**Comment by 5y** [comment]
@ThorstenBux Do you have any idea?
![image](https://user-images.githubusercontent.com/1941933/37048575-851ea7a8-2183-11e8-8cad-87c2b307e438.png)

**Comment by ThorstenBux** [comment]
https://archive.artoolkit.org/documentation/doku.php?id=4_Android:android_native

that might help you go further. It explains how to build the native libraries.
